File: sesac7_flutter_recipe_app/lib/presentation/component/button/filter_button.dart
import 'package:flutter/material.dart';
import 'package:flutter_recipe_app/ui/app_colors.dart';
import 'package:flutter_recipe_app/ui/text_styles.dart';

class FilterButton extends StatelessWidget {
  final String filterTitle;
  final bool isSelected;
  final Function(bool isSelected) onValueChange;

  const FilterButton({
    super.key,
    required this.filterTitle,
    required this.isSelected,
    required this.onValueChange,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        onValueChange(!isSelected);
      },
      child: Container(
        padding: const EdgeInsets.symmetric(horizontal: 10.0, vertical: 5.0),
        decoration: BoxDecoration(
          border: Border.fromBorderSide(
            BorderSide(
              color: isSelected ? AppColors.primary100 : AppColors.primary80,
              width: 1.0,
            ),
          ),
          borderRadius: BorderRadius.circular(10.0),
          color: isSelected ? AppColors.primary100 : AppColors.white,
        ),
        child: Text(
          filterTitle,
          style: isSelected
              ? TextStyles.selectedFilterTitle
              : TextStyles.unSelectedFilterTitle,
        ),
      ),
    );
  }
}

File: compose_recipe/app/src/main/java/com/example/recipeapp/presentation/component/button/FilterButton.kt
package com.example.recipeapp.presentation.component.button

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.example.recipeapp.ui.theme.Primary100
import com.example.recipeapp.ui.theme.Primary80
import com.example.recipeapp.ui.theme.White
import com.example.recipeapp.ui.theme.selectedFilterTitle
import com.example.recipeapp.ui.theme.unSelectedFilterTitle

@Composable
fun FilterButton(
    modifier: Modifier = Modifier,
    filterTitle: String,
    isSelected: Boolean,
    onValueChange: (Boolean) -> Unit
) {
    Box(
        modifier = modifier
            .height(28.dp)
            .border(
                width = 1.dp,
                color = if (isSelected) Primary100 else Primary80,
                shape = RoundedCornerShape(size = 10.dp)
            )
            .background(
                color = if (isSelected) Primary100 else White,
                shape = RoundedCornerShape(size = 10.dp)
            )
            .clickable {
                onValueChange(!isSelected)
            },
        contentAlignment = Alignment.Center
    ) {
        Text(
            modifier = Modifier
                .padding(vertical = 5.dp, horizontal = 10.dp),
            text = filterTitle,
            style = if (isSelected) selectedFilterTitle else unSelectedFilterTitle
        )
    }
}

@Preview
@Composable
fun FilterButtonPreview() {
    var isSelected by remember { mutableStateOf(false) }

    FilterButton(
        filterTitle = "Text",
        isSelected = isSelected
    ) {
        isSelected = it
    }
}
